### ISSUE TITLE
Improve Range usability

### ISSUE THREAD

**Issue by mfelsche** [ponylang/rfcs#109]
Current issues with `collections.Range` that hurt its usability:

* negative Ranges (where `min > max`) is only expressable for signed integers:
  ```pony
  // this is not possible, as the inc parameter is also U8
  Range[U8](10, 0, -1)
  ```

* The default `inc` should only be `1` if `min <= max`, otherwise it should be `-1` to ensure that we iterate into the expected direction:

  ```pony
   // this should infer a default step of -1
  Range[U8](10, 0)

  // this should infer a default step of 1
  Range[U64](0, 10)
  ```

**Comment by EpicEric** [comment]
Solved by ponylang/ponyc#2350

**Comment by SeanTAllen** [comment]
Thanks @EpicEric!
